dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -163,35 +162,29 @@ internal KernelFunction(string name, string? pluginName, string description, IRe
 
         TagList tags = new() { { MeasurementFunctionTagName, this.Name } };
         long startingTimestamp = Stopwatch.GetTimestamp();
-        FunctionResult? functionResult = null;
+        FunctionResult? functionResult = new(this);
         try
         {
             // Quick check for cancellation after logging about function start but before doing any real work.
             cancellationToken.ThrowIfCancellationRequested();
 
-            // Invoke pre-invocation filter. If it requests cancellation, throw.
-            var invokingContext = kernel.OnFunctionInvokingFilter(this, arguments);
-
-            if (invokingContext?.Cancel is true)
+            var invocationContext = await kernel.OnFunctionInvocationAsync(this, arguments, async (context) =>
             {
-                throw new OperationCanceledException("A function filter requested cancellation before function invocation.");
-            }
+                // Invoke the function.
+                functionResult = await this.InvokeCoreAsync(kernel, context.Arguments, cancellationToken).ConfigureAwait(false);
 
-            // Invoke the function.
-            functionResult = await this.InvokeCoreAsync(kernel, arguments, cancellationToken).ConfigureAwait(false);
+                // Update context with result.
+                context.Result = functionResult;
+            }).ConfigureAwait(false);
 
-            // Invoke the post-invocation filter. If it requests cancellation, throw.
-            var invokedContext = kernel.OnFunctionInvokedFilter(arguments, functionResult);
-
-            if (invokedContext is not null)
+            if (invocationContext is not null)
```

Comment:
Why is this null check needed?

Reply:
Same answer as above, `null` context means that there were no filters triggered during function execution. This allows us to understand whether we should override final function result or just return it as it is.